### PR TITLE
Support new training/inference PR builds

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Add new PR builds to dev toml
- Remove top level buildspec files; split "SM Lite" as well
- Add new condition to dev toml to modulate training/inference building
- Update PR template
- Remove change-set logic. Now, all builds are executed explicitly by the user via toml

### Tests
- [x] test TF 2.9 inference [f2db16d](https://github.com/aws/deep-learning-containers/pull/2637/commits/f2db16db22ed20eb753cc1edbaeb30fdc3062235)
- [x] test TF 2.9 inference and TF 2.11 training [7cca19b](https://github.com/aws/deep-learning-containers/pull/2637/commits/7cca19bdcbf728d935cbb9426c90d83b6940aae8)
- [x] test pytorch inference neuron [c63095a](https://github.com/aws/deep-learning-containers/pull/2637/commits/c63095abbaa51e3b1e888d2f0f6ce61700ad99e0)
- [x] test pytorch training neuron [c63095a](https://github.com/aws/deep-learning-containers/pull/2637/commits/c63095abbaa51e3b1e888d2f0f6ce61700ad99e0)
- [ ] test PT 1.12 training [a527912](https://github.com/aws/deep-learning-containers/pull/2637/commits/a52791248d3395bf91a635872cdd98d9c41ca8b0)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -40,16 +40,25 @@ def main():
...
     ei_build_mode = parse_dlc_developer_configs("dev", "ei_mode")
     neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
     graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
     habana_build_mode = parse_dlc_developer_configs("dev", "habana_mode")
     trcomp_build_mode = parse_dlc_developer_configs("dev", "trcomp_mode")
     hf_trcomp_build_mode = parse_dlc_developer_configs("dev", "huggingface_trcomp_mode")
 
+    # Condition to check whether we are training or inference dedicated/enabled
+
+    train_or_inf_enabled = (training_dedicated and training_enabled) or (inference_dedicated and inference_enabled)
```
Comment (arjkesh) [inline src/main.py:60]: may want to add condition to check for "" to still support older types of PR jobs if we want to bring them back in any capacity

File: src/main.py [hunk 2]
```diff
@@ -40,16 +40,25 @@ def main():
...
+    training_enabled = parse_dlc_developer_configs("build", "build_training")
+    inference_enabled = parse_dlc_developer_configs("build", "build_inference")
     ei_build_mode = parse_dlc_developer_configs("dev", "ei_mode")
     neuron_build_mode = parse_dlc_developer_configs("dev", "neuron_mode")
     graviton_build_mode = parse_dlc_developer_configs("dev", "graviton_mode")
     habana_build_mode = parse_dlc_developer_configs("dev", "habana_mode")
     trcomp_build_mode = parse_dlc_developer_configs("dev", "trcomp_mode")
     hf_trcomp_build_mode = parse_dlc_developer_configs("dev", "huggingface_trcomp_mode")
 
+    # Condition to check whether we are training or inference dedicated/enabled
```
Comment (tejaschumbalkar) [inline src/main.py:58]: nit: remove `we`

**Review by tejaschumbalkar — COMMENTED** [review]
LGTM

**Review by tejaschumbalkar — APPROVED** [review]

**Review by tejaschumbalkar — COMMENTED** [review]
changeset changes are good

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**